src/aiff.rs

```diff
@@ -0,0 +1,112 @@
+use std::fs;
+use std::fs::File;
+use std::io::prelude::*;
+use std::io::SeekFrom;
+use std::path::Path;
+use crate::{Error, ErrorKind};
+use crate::{Tag, Version};
+
+pub fn load_aiff_id3(path: impl AsRef<Path>) -> crate::Result<Tag> {
+    let mut file = File::open(path)?;
+    loop {
+        // Read chunk ID
+        let mut chunk_id: [u8; 4] = [0; 4];
+        // EOF
+        if file.read(&mut chunk_id)? == 0 {
+            break;
+        }
+
+        // Read chunk size
+        let mut chunk_size_raw: [u8; 4] = [0; 4];
+        file.read(&mut chunk_size_raw)?;
+        let chunk_size = u32::from_be_bytes(chunk_size_raw);
+
+        // Skip FORM chunk type, get its chunks
+        if &chunk_id == b"FORM" {
+            file.seek(SeekFrom::Current(4))?;
+            continue;
+        }
+
+        if &chunk_id[0..3] == b"ID3" {
+            return Ok(Tag::read_from(file.take(chunk_size as u64))?);
+        }
+
+        file.seek(SeekFrom::Current(chunk_size as i64))?;
+    }
+
+    Err(Error::new(ErrorKind::NoTag, "No tag chunk found!"))
+}
+
+// Wrapper to delete temp file
+pub fn overwrite_aiff_id3(path: impl AsRef<Path>, tag: &Tag, version: Version) -> crate::Result<()> {
+    let res = overwrite_aiff_id3_raw(&path, tag, version);
+    if res.is_err() {
+        let new_path = format!("{}.ID3TMP", path.as_ref().to_str().unwrap());
```

Comment:
You should use one of the methods on `Path`, instead of manually working on it as if it was a String. Your current implementation would panic for paths which aren't valid utf-8, because of the call to `to_str().unwrap()`.